lib/rules/no-invalid-double-slash-comments/index.mjs

```diff
@@ -36,6 +37,8 @@ const rule = (primary) => {
 		root.walkRules((ruleNode) => {
 			const selectors = getRuleSelector(ruleNode).split(',');
 
+			assert(ruleNode.source);
```

Comment:
@romainmenke can I assume that `source` is always present?

Reply:
No, `source` is correctly typed and can be absent.
Impossible to test with `testRule` as this does ensure that `source` is present, but other plugins with rules might insert nodes without `source` or other runners might not provide it.

Reply:
> but other plugins with rules might insert nodes without source or other runners might not provide it.

More concretely is `ruleStringified ??= ruleNode.source.input.css;` safe to use?
If not, what would be the fallback?

Reply:
I am not sure, I would have to investigate and I don't currently have time :/

In general I know that `source` is in reference to the source CSS (as a string) that was used to construct the AST.

When manually creating AST nodes through the programmatic API's there is no source CSS as a string for those nodes.

So you should create some AST nodes programmatically and then experiment from there.